gene/database/postgresql.py

```diff
@@ -0,0 +1,867 @@
+    ) -> Optional[Dict]:
+        """Retrieve normalized record from DB.
+
+        :param concept_id: normalized ID for the merged record
+        :param case_sensitive:
+        :return: normalized record if successful
+        """
+        concept_id = concept_id.lower()
+        query = "SELECT * FROM gene_merged WHERE lower(concept_id) = %s;"
+        with self.conn.cursor() as cur:
+            cur.execute(query, [concept_id])
+            result = cur.fetchone()
+        if not result:
+            return None
+
+        merged_record = {
+            "concept_id": result[0],
+            "symbol": result[1],
+            "symbol_status": result[2],
+            "previous_symbols": result[3],
+            "label": result[4],
+            "strand": result[5],
+            "ensembl_locations": result[6],
+            "hgnc_locations": result[7],
+            "ncbi_locations": result[8],
+            "location_annotations": result[9],
+            "ensembl_biotype": result[10],
+            "hgnc_locus_type": result[11],
+            "ncbi_gene_type": result[12],
+            "aliases": result[13],
+            "associated_with": result[14],
+            "xrefs": result[15],
+            "item_type": "merger",
+        }
+        return {k: v for k, v in merged_record.items() if v}
+
+    def get_record_by_id(self, concept_id: str, case_sensitive: bool = True,
+                         merge: bool = False) -> Optional[Dict]:
+        """Fetch record corresponding to provided concept ID
+        :param str concept_id: concept ID for gene record
+        :param bool case_sensitive:
+        :param bool merge: if true, look for merged record; look for identity
+            record otherwise.
+        :return: complete gene record, if match is found; None otherwise
+        """
+        if merge:
+            return self._get_merged_record(concept_id, case_sensitive)
+        else:
+            return self._get_record(concept_id, case_sensitive)
+
+    def get_refs_by_type(self, query: str, match_type: str) -> List[str]:
+        """Retrieve concept IDs for records matching the user's query. Other methods
+        are responsible for actually retrieving full records.
+
+        :param query: string to match against
+        :param match_type: type of match to look for. Should be one of {"symbol",
+            "prev_symbol", "alias", "xref", "associated_with"} (use `get_record_by_id`
+            for concept ID lookup)
+        :return: list of associated concept IDs. Empty if lookup fails.
+        """
+        if match_type == "symbol":
+            table = "gene_symbols"
+        elif match_type == "prev_symbol":
+            table = "gene_previous_symbols"
+        elif match_type == "alias":
+            table = "gene_aliases"
+        elif match_type == "xref":
+            table = "gene_xrefs"
+        elif match_type == "associated_with":
+            table = "gene_associations"
+        else:
+            raise ValueError
```

Comment:
Should we include a descriptive error msg?

Reply:
👍 yeah we should make sure this gets caught and uses enums and whatnot